newsletter-bundle/src/Event/SendNewsletterEvent.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsletterBundle\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+class SendNewsletterEvent extends Event
+{
+    private string $recipientAddress;
+    private string $text;
+    private string $html;
+    private bool $submissionPrevented = false;
+    private bool $htmlAllowed = true;
+    private array $recipientData = [];
+    private array $newsletterData = [];
+
+    public function __construct(string $recipientAddress, string $text, string $html = '')
+    {
+        $this->recipientAddress = $recipientAddress;
+        $this->text = $text;
+        $this->html = $html;
+    }
+
+    public function getRecipientAddress(): string
+    {
+        return $this->recipientAddress;
+    }
+
+    public function setRecipientAddress(string $recipientAddress): self
+    {
+        $this->recipientAddress = $recipientAddress;
+
+        return $this;
+    }
+
+    public function isSubmissionPrevented(): bool
```

Comment:
How do you like `isSkipSending()` and `setSkipSending()` instead?

Reply:
How about `discardNewsletter` or `preventSending`? Similar to `stopPropagation()`. Also, this should probably call `stopPropagation()`?

Reply:
`isPreventSending` and `setPreventSending` sounds good to me. I have renamed the property and methods in https://github.com/contao/contao/pull/3489/commits/69e6b717477f6d09728b16c09b884ec14ea52e20

I don't think the event should call `stopPropagation()` automatically – this should be a conscious decision by the developer. There might be cases where one listener prevents the sending while another listener wants to explicitly re-enable some prevented newsletters.